common/transport/amqp/src/amqp.ts

```diff
@@ -132,6 +88,185 @@ export class Amqp {
         forever: false
       }
     }, amqp10.Policy.EventHub));
+
+    this._fsm = new machina.Fsm({
+      initialState: 'disconnected',
+      states: {
+        disconnected: {
+          connect: (connectCallback) => {
+            this._fsm.transition('connecting');
+            let connectError = null;
+            const connectErrorHandler = (err) => {
+              connectError = err;
+            };
+
+            const amqpErrorHandler = (err) => {
+              debug('amqp10 error: ' + err.toString());
+              if (this._disconnectHandler) {
+                this._disconnectHandler(err);
+              }
+            };
+
+            this._amqp.on(_amqpClientError, connectErrorHandler);
+            this._amqp.connect(this.uri)
```

Comment:
we step into the connecting state in the connect function

---
In reply to: [128124325](https://github.com/Azure/azure-iot-sdk-node/pull/55#discussion_r128124325) [](ancestors = 128124325)